src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -1091,7 +1090,6 @@ static int client_receive_advertise(sd_dhcp6_client *client, DHCP6Message *adver
 
         if (r < 0 || pref_advertise > pref_lease) {
                 client_set_lease(client, lease);
-                lease = NULL;
```

Comment:
This is correct.

Another way would be to make `client_set_lease` take ownership of the reference. I would prefer that.